lib/src/auth/auth.dart

```diff
@@ -29,7 +29,7 @@ abstract class BaseAuthenticator {
   auth.AccessToken _accessToken;
   String _lastUri;
 
-  Future authenticate(Map<String, String> metadata, String uri) async {
+  Future<Null> authenticate(Map<String, String> metadata, String uri) async {
```

Comment:
Can't you use `Future<void>` here yet?

Reply:
I'd like to keep this code running on latest stable Dart SDK as long as possible, so not yet...